### DISCUSSION TITLE
odd zoom with pointcloudlayer and orbitview

### DISCUSSION THREAD

**Discussion by acthp** [visgl/deck.gl#6696] [Q&A] [answered]
Point size scales differently with zoom depending on camera direction. Any idea what would cause this?

![deckzoom](https://user-images.githubusercontent.com/494506/155590656-01305098-a450-4200-983f-ad50e20aba65.gif)

**Comment by ethanruiyu** [comment, marked as answer]
<img width="810" alt="image" src="https://user-images.githubusercontent.com/81229588/155952195-ef9030e5-780f-4bc0-800e-cd4a1dfdac02.png">


<img width="707" alt="image" src="https://user-images.githubusercontent.com/81229588/155952062-d9cba2b7-b41a-4f5a-950a-19c59d11e3d1.png">

**Reply by acthp** [reply]
Setting sizeUnits to 'meters' seems to fix the problem, though I still don't understand what the zoom is trying to do in the 'pixels' and 'common' modes.

**Comment by Pessimistress** [comment]
If you log `viewState.target` in `onViewStateChange` you can see that as you pan, the camera target has moved to somewhere far away from the point cloud. Then when you rotate around that target in the middle of nowhere, the point cloud can move very close to the camera, causing distortions. 

The OrbitView itself does not know about the boundaries of your data. You can add constraints to the controller using this technique: https://deck.gl/docs/developer-guide/interactivity#add-constraints-to-view-state

[example](https://codepen.io/Pessimistress/pen/ZEaqPZE)
